### PR TITLE
feat(Calibration Tool): add a calibration tool that can manually recalibrate an image and measurements on it

### PR DESCRIPTION
**Pull request by md-prog**
### Context

Manual Calibration Tool implementation.
A DICOM image usually have calibration information (or pixel spacing information) that represents the mapping of real-world size and the pixels in the image. Sometimes this information is not available and even when it is available, we may want to manually re-calibrate the image.

#### UX

- Similar to the line(length) tool
- but once the line is drawn, a popup shows up to collect input about the physical distance represented by the line.
- Click OK, then the viewport and all the measurements will be updated

#### Notes

- The re-calibration performed by this tool would be temporary and won't be persisted/uploaded back to the source image yet.

#### PR

- [x] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [x] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [x] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] OS: Windows 10
- [x] Node version: 16.15.0
- [x] Browser: Chrome 109.0.5414.121 (Official Build) (64-bit)

### CODE REVIEW

File: extensions/cornerstone/src/initMeasurementService.js [hunk 1]
```diff
@@ -66,14 +68,25 @@ const initMeasurementService = (
...
 
+  // On the UI side, the Calibration Line tool will work almost the same as the
+  // Length tool
+  MeasurementService.addMapping(
+    csTools3DVer1MeasurementSource,
+    'CalibrationLine',
+    Length.matchingCriteria,
+    Length.toAnnotation,
+    Length.toMeasurement
+  );
```
Comment (sedghi) [inline extensions/cornerstone/src/initMeasurementService.js:79]: I don't understand why you are adding it to the measurementService? The tools added here store state in the measurementService, and need mappings. But to me, the calibration tool is a tool that runs a call back on finish. Do we really need mapping here?

File: extensions/cornerstone/src/initMeasurementService.js [hunk 2]
```diff
@@ -95,15 +108,35 @@ const connectToolsToMeasurementService = (
...
+            },
+            () => true
+          )
+          .finally(() => {
+            // we don't need the calibration line lingering around, remove the
+            // annotation from the display
+            removeAnnotation(annotationUID);
+            removeMeasurement(csToolsEvent);
+            // this will ensure redrawing of annotations
+            CornerstoneViewportService.resize();
```
Comment (sedghi) [inline extensions/cornerstone/src/initMeasurementService.js:126]: Yeah I thought you would do this, so we don't need the measurement service mapping.

File: extensions/cornerstone/src/tools/CalibrationLineTool.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import getActiveViewportEnabledElement from '../utils/getActiveViewportEnabledElement';
+
+/**
+ * Calibration Line tool works almost the same as the
+ */
+class CalibrationLineTool extends LengthTool {
+  static toolName = 'CalibrationLine';
+
+  _renderingViewport: any;
+  _lengthToolRenderAnnotation = this.renderAnnotation;
```
Comment (sedghi) [inline extensions/cornerstone/src/tools/CalibrationLineTool.ts:14]: interesting

**Comment by sedghi** [comment]
See my comment. I don't think you need to add it to measurement service since it is not getting displayed in the measurement panel, nore getting exported right?

File: extensions/cornerstone/src/initMeasurementService.js (hunk 1 above)
Comment (md-prog) [reply extensions/cornerstone/src/initMeasurementService.js:79]: You're right, the code is not necessary

**Comment by arturojain** [comment]
What OHIF version does this PR support?

**Review by sedghi — APPROVED** [review]
This looks good to me, please rebase and we can merge

**Comment by md-prog** [comment]
@sedghi I am sorry for the delay. Rebased, fixed (some code changes required after rebase), and tested.

**Pull request merged**